dali/operators/audio/resample.cc

```diff
@@ -0,0 +1,168 @@
+// Copyright (c) 2022, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <map>
+#include <vector>
+#include "dali/operators/audio/resample.h"
+#include "dali/operators/audio/resampling_params.h"
+#include "dali/kernels/kernel_params.h"
+
+namespace dali {
+
+DALI_SCHEMA(experimental__AudioResample)
+  .DocStr(R"(Resamples a signal with a different sampling rate.
+
+The resampling is achieved by applying a sinc filter with Hann window with an extent
+controlled by the ``quality`` argument.
+
+The resampling ratio can specified either directly or as a ratio of target to source sampling rate
+or calculated from the ratio of requested output length to input length.
```

Comment:
```suggestion
The resampling ratio can specified either directly, or as a ratio of target to source sampling rate,
or calculated from the ratio of requested output length to input length.
```

Reply:
Actually, the rules for comma before "or" are more complex - there should be one if an _independent clause_ follows or.

> The resampling ratio can be specified directly, or it can be calculated as a ratio of input and output sampling rates.

Here, we have two independent clauses ("[ratio] can be be specified..." and "it can be calculated...").
In the original text, however, the clauses are dependent - they both refer to "be specified": "directly or as a ratio...".

Having said all that, the sentence is indeed incorrect, because:
- it's missing the verb
- it uses "either" in front of multiple choice

I think it should be:
> The resampling ratio can be specified directly or as a ratio of target to source sampling rate, or calculated...

but perhaps the second comma is not mandatory, as the last clause can be treated as dependent wrt "be" (specified or calculated).

Reply:
Up to you